python-frontend/src/main/java/org/sonar/python/index/DescriptorUtils.java

```diff
@@ -67,11 +67,57 @@ public static Symbol symbolFromDescriptor(Descriptor descriptor, ProjectLevelSym
           .map(a -> DescriptorUtils.symbolFromDescriptor(a, projectLevelSymbolTable, symbolName, createdSymbolsByDescriptor, new HashMap<>()))
           .collect(Collectors.toSet()));
         return ambiguousSymbol;
+      case ALIAS:
+        Descriptor recreatedDescriptor = recreateDescriptorFromAlias((AliasDescriptor) descriptor);
+        return symbolFromDescriptor(recreatedDescriptor, projectLevelSymbolTable, symbolName, createdSymbolsByDescriptor, createdSymbolsByFqn);
       default:
         throw new IllegalStateException(String.format("Error while creating a Symbol from a Descriptor: Unexpected descriptor kind: %s", descriptor.kind()));
     }
   }
 
+  private static Descriptor recreateDescriptorFromAlias(AliasDescriptor aliasDescriptor) {
+    return switch (aliasDescriptor.originalDescriptor().kind()) {
```

Comment:
if replace this `kind` `switch-case` with just `instanceof` `if-else` - it will allow to get rid of unsafe class casts in the `recreateFunctionDescriptor` and the `recreateClassDescriptor` methods like:
```java
private static Descriptor recreateDescriptorFromAlias(AliasDescriptor aliasDescriptor) {
  if (aliasDescriptor.originalDescriptor() instanceof FunctionDescriptor functionDescriptor) {
    return recreateFunctionDescriptor(aliasDescriptor, functionDescriptor);
  } else if (aliasDescriptor.originalDescriptor() instanceof ClassDescriptor classDescriptor) {
    return recreateClassDescriptor(aliasDescriptor, classDescriptor);
  }
  throw new IllegalStateException(String.format("Error while recreating a descriptor from an alias: Unexpected alias kind: %s", aliasDescriptor.kind()));
}
```